packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -141,22 +331,31 @@ export default declare((api, options) => {
           }
         }
 
+        // Transform private props before publics.
+        const privates = [];
+        const privateInits = [];
         for (const prop of props) {
-          const propNode = prop.node;
-          if (propNode.decorators && propNode.decorators.length > 0) continue;
+          const inits = [];
+          privateInits.push(inits);
```

Comment:
Nit: this two lines can be moved inside the if statement.

Also, why can't these loops be merged?

Reply:
> Also, why can't these loops be merged?

The privates need to be transformed first in case a public field references one. I should probably test that a private field that references another transforms correctly...